src/rest.ts

```diff
@@ -162,10 +163,31 @@ const dandiRest = new Vue({
       const response = await client.get('dandisets/', { params });
       return response;
     },
-    async createDandiset(name: string, description: string): Promise<AxiosResponse<Dandiset>> {
-      const { schema_version } = await this.info();
-      const metadata = { name, description, schemaVersion: schema_version };
-      return client.post('dandisets/', { name, metadata });
+    async createDandiset(
+      name: string, description: string, embargoed: boolean, awardNumber: IdentifierForAnAward,
+    ): Promise<AxiosResponse<Dandiset>> {
```

Comment:
Generally speaking, I think creating new functions like `createEmbargoedDandiset` is a better design pattern than adding switch arguments like `embargoed` to existing functions. More complicated functions are less testable and harder to read, and adding more arguments scales complexity exponentially.

In this case there is some duplicated code, so to keep things DRY perhaps `createDandiset(...)` could take optional `metadata` and `params` arguments that are empty if unspecified. Then `createEmbargoedDandiset` would call `createDandiset` with appropriately initialized `metadata` and `params`.

Reply:
Also, I couldn't pick just one, but this is worth [thinking about](https://www.google.com/search?q=don%27t+use+boolean+parameters&oq=dont+use+boolean+parameters).